pkg/utils/dataset.go

```diff
@@ -71,6 +77,21 @@ func GetAccessModesOfDataset(client client.Client, name, namespace string) (acce
 
 }
 
+func GetPVCStorageCapacityOfDataset(client client.Client, name, namespace string) (storageCapacity resource.Quantity, err error) {
+	dataset, err := GetDataset(client, name, namespace)
+	if err != nil {
+		return storageCapacity, err
```

Comment:
Enhance error context by using `fmt.Errorf("failed to get dataset %s/%s: %w", namespace, name, err)`.